### PR TITLE
core: ardupilot manager: add run permission to navigator binary

### LINKED ISSUE #303: Navigator Ardusub binary is not being chmodded +x ?
I removed the binary to trigger a re-download.
This was the result in the autopilot session:
image

it worked after I manually chmodded the file

### PR DESCRIPTION
**Pull request by Williangalvani**
this should fix #303 
needs testing.

### CODE REVIEW

File: core/services/ardupilot_manager/firmware_download/FirmwareDownload.py [hunk 1]
```diff
@@ -232,7 +232,9 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
             Optional[pathlib.Path]: Temporary path for the firmware file, None if unable to download or validate file.
         """
         if platform == Platform.Navigator and vehicle == Vehicle.Sub:
-            return FirmwareDownload._download_navigator()
+            path = FirmwareDownload._download_navigator()
+            os.chmod(str(path), stat.S_IXOTH)
+            return path
```
Comment (rafaellehmkuhl) [inline core/services/ardupilot_manager/firmware_download/FirmwareDownload.py:237]: Oh, good one. This was introduced by my SITL PR (#289).

Line 283 should chmod the binary, as its .elf, but as it's a navigator one it returns before.

Maybe we could merge both?

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

File: core/services/ardupilot_manager/firmware_download/FirmwareDownload.py (hunk 1 above)
Comment (rafaellehmkuhl) [reply core/services/ardupilot_manager/firmware_download/FirmwareDownload.py:237]: In fact, as the Navigator download is a temporary solution, I'm very ok with it.

**Review by rafaellehmkuhl — APPROVED** [review]

File: core/services/ardupilot_manager/firmware_download/FirmwareDownload.py (hunk 1 above)
Comment (patrickelectric) [reply core/services/ardupilot_manager/firmware_download/FirmwareDownload.py:237]: Check #306

**Comment by Williangalvani** [comment]
superseeded by #306

**Pull request closed without merging**